src/flightRecorder.cpp

```diff
@@ -1068,6 +1214,41 @@ class Recording {
         buf->putVar32(start, buf->offset() - start);
     }
 
+    void recordProcessSample(Buffer* buf, const ProcessInfo* info, u64 sampling_time) {
+        int start = buf->skip(5);
+        buf->put8(T_PROCESS_SAMPLE);
+        buf->putVar64(sampling_time);
```

Comment:
Event time is supposed to be in TSC ticks (and that's how JFR Metadata describes it), but you pass nanoseconds here.